### PR TITLE
Added pose conversions for 2d transformations.

### PR DESCRIPTION
**Pull request by fabianbl**

### CODE REVIEW

**Review by mfehr — APPROVED** [review]
lgtm

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 1]
```diff
@@ -37,46 +39,89 @@ void quaternionMsgToKindr(
...
   quaternionMsgToEigen(msg, kindr_double);
   *kindr = kindr_double.cast<Scalar>();
 }
 
+template <typename Scalar>
+void rotationKindr2DToMsg(
+    const Eigen::Rotation2D<Scalar>& kindr, geometry_msgs::Quaternion* msg) {
+  CHECK_NOTNULL(msg);
+  const kindr::minimal::RotationQuaternionTemplate<Scalar> quat(
+      kindr::minimal::AngleAxisTemplate<Scalar>(kindr.angle(), 0.0, 0.0, 1.0));
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h]: shouldn't the numbers be casted to scalar?

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 2]
```diff
@@ -37,46 +39,89 @@ void quaternionMsgToKindr(
...
+  quaternionEigenToMsg(quat.toImplementation(), *msg);
+}
+
+template <typename Scalar>
+void quaternionMsgToKindr2D(
+    const geometry_msgs::Quaternion& msg, Eigen::Rotation2D<Scalar>* kindr) {
+  CHECK_NOTNULL(kindr);
+  CHECK_LT(std::abs(msg.x), std::numeric_limits<Scalar>::epsilon())
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h]: msg.x is double so you need double epsilon here I think

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 3]
```diff
@@ -37,46 +39,89 @@ void quaternionMsgToKindr(
...
   pointMsgToEigen(msg, *kindr);
 }
 
 template <typename Scalar>
-void vectorKindrToMsg(const Eigen::Matrix<Scalar, 3, 1>& kindr,
-                      geometry_msgs::Vector3* msg) {
+void pointKindr2DToMsg(
+    const Eigen::Matrix<Scalar, 2, 1>& kindr, geometry_msgs::Point* msg) {
+  CHECK_NOTNULL(msg);
+  msg->x = kindr.x();
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h]: cast to double?

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 4]
```diff
@@ -37,46 +39,89 @@ void quaternionMsgToKindr(
...
+  msg->y = kindr.y();
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h]: cast to double?

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 5]
```diff
@@ -37,46 +39,89 @@ void quaternionMsgToKindr(
...
+}
+
+template <typename Scalar>
+void pointMsgToKindr2D(
+    const geometry_msgs::Point& msg, Eigen::Matrix<Scalar, 2, 1>* kindr) {
+  CHECK_NOTNULL(kindr);
+  // Verify that we got a proper 2D pose.
+  CHECK_LT(std::abs(msg.z), std::numeric_limits<Scalar>::epsilon())
+      << "No proper 2D position.";
+  kindr->x() = msg.x;
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h]: cast to Scalar?

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 6]
```diff
@@ -37,46 +39,89 @@ void quaternionMsgToKindr(
...
+  kindr->y() = msg.y;
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h]: cast to Scalar?

**Review by dymczykm — APPROVED** [review]
scolgtm

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h [hunk 7]
```diff
@@ -37,46 +39,91 @@ void quaternionMsgToKindr(
...
+void quaternionMsgToKindr2D(
+    const geometry_msgs::Quaternion& msg, Eigen::Rotation2D<Scalar>* kindr) {
+  CHECK_NOTNULL(kindr);
+  CHECK_LT(std::abs(msg.x), std::numeric_limits<double>::epsilon())
+      << "No proper 2D rotation.";
+  CHECK_LT(std::abs(msg.y), std::numeric_limits<double>::epsilon())
+      << "No proper 2D rotation.";
+  kindr::minimal::RotationQuaternionTemplate<Scalar> quat;
+  quaternionMsgToEigen(msg, quat.toImplementation());
+  const kindr::minimal::AngleAxisTemplate<Scalar> angle_axis(quat);
```
Comment (dymczykm) [inline minkindr_conversions/include/minkindr_conversions/kindr_msg.h:78]: I think you need `quat.cast<Scalar>()`

**Review by dymczykm — APPROVED** [review]

File: minkindr_conversions/include/minkindr_conversions/kindr_msg.h (hunk 7 above)
Comment (fabianbl) [reply minkindr_conversions/include/minkindr_conversions/kindr_msg.h:78]: @dymczykm but isn't this already `Scalar`?

Comment (dymczykm) [reply minkindr_conversions/include/minkindr_conversions/kindr_msg.h:78]: nvmnd

**Pull request merged**